dubbo-remoting/dubbo-remoting-http12/src/main/java/org/apache/dubbo/remoting/http12/message/LengthFieldStreamingDecoder.java

```diff
@@ -130,16 +127,12 @@ private void deliver() {
     }
 
     private void processHeader() throws IOException {
-        ByteArrayOutputStream bos = new ByteArrayOutputStream(lengthFieldOffset + lengthFieldLength);
         byte[] offsetData = new byte[lengthFieldOffset];
         int ignore = accumulate.read(offsetData);
-        bos.write(offsetData);
         processOffset(new ByteArrayInputStream(offsetData), lengthFieldOffset);
         byte[] lengthBytes = new byte[lengthFieldLength];
         ignore = accumulate.read(lengthBytes);
-        bos.write(lengthBytes);
         requiredLength = bytesToInt(lengthBytes);
-        this.dataHeader = new ByteArrayInputStream(bos.toByteArray());
 
         // Continue reading the frame body.
         state = DecodeState.PAYLOAD;
```

Comment:
**Security Issue**: The removal of ByteArrayOutputStream usage and direct passing of byte arrays without validation can lead to security vulnerabilities related to improper input validation. <br> **Fix**: Validate the byte arrays for expected format and size before processing to mitigate potential buffer overflow or data corruption vulnerabilities. <br> **Code Suggestion**: 
 ```
 byte[] offsetData = new byte[lengthFieldOffset];
 int ignore = accumulate.read(offsetData);
 + if (!validateByteArray(offsetData)) {
 +     throw new InvalidDataException("Invalid byte array format");
 + }
 processOffset(new ByteArrayInputStream(offsetData), lengthFieldOffset);
 ```